src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.cc

```diff
@@ -0,0 +1,396 @@
+  auto actuatorMsgComp =
+      _ecm.Component<components::Actuators>(this->model.Entity());
+
+  // Actuators messages can come in from transport or via a component. If a
+  // component is available, it takes precedence.
+  if (actuatorMsgComp)
+  {
+    msg = actuatorMsgComp->Data();
+  }
+  else
+  {
+    std::lock_guard<std::mutex> lock(this->recvdActuatorsMsgMutex);
+    if (this->recvdActuatorsMsg.has_value())
+    {
+      msg = *this->recvdActuatorsMsg;
+      this->recvdActuatorsMsg.reset();
+    }
+  }
+
+  if (msg.has_value())
+  {
+    if (this->actuatorNumber > msg->normalized_size() - 1)
+    {
+      gzerr << "You tried to access index " << this->actuatorNumber
+        << " of the Actuator array which is of size "
+        << msg->velocity_size() << std::endl;
+      return;
+    }
+  }else{
+    return;
+  }
+
+  // Process input
+  double ref_duty_cycle_ = msg->normalized(this->actuatorNumber);
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Ref duty cycle: " << ref_duty_cycle_ << std::endl;
+
+  // Calculate cycle start time
+  if (this->samplingTime >= 1.0/this->dutyCycleFrequency) {
+    if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Cycle completed. Resetting cycle start time." << std::endl;
+    this->cycleStartTime = this->simTime;
+  }
+
+  // Calculate sampling time instant within the cycle
+  this->samplingTime = this->simTime - this->cycleStartTime;
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": PWM Period: " << 1.0/this->dutyCycleFrequency << " Cycle Start time: " << this->cycleStartTime << " Sampling time: " << this->samplingTime << std::endl;
+
+  double ref_duty_cycle = ref_duty_cycle_ * (1.0 / this->dutyCycleFrequency);
+  double force = this->samplingTime <= ref_duty_cycle ? this->maxThrust : 0.0;
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Force: " << force << "  Sampling time: " << this->samplingTime << "  Ref duty cycle: " << ref_duty_cycle_ << std::endl;
+  
+  Link link(this->linkEntity);
+  const auto worldPose = link.WorldPose(_ecm);
+
+  // Apply a force to the link.
+  link.AddWorldForce(_ecm,
+                      worldPose->Rot().RotateVector(math::Vector3d(0, 0, force)));
+  if (DEBUG && this->actuatorNumber == 0) std::cout << this->actuatorNumber << ": Value: " << msg->normalized(this->actuatorNumber) << 
+               "  Force: " << force << std::endl;
+}
+
+GZ_ADD_PLUGIN(SpacecraftThrusterModel,
+                    System,
+                    SpacecraftThrusterModel::ISystemConfigure,
+                    SpacecraftThrusterModel::ISystemPreUpdate)
+
+GZ_ADD_PLUGIN_ALIAS(SpacecraftThrusterModel,
+                          "gz::sim::systems::SpacecraftThrusterModel")
+
+// TODO(CH3): Deprecated, remove on version 8
+GZ_ADD_PLUGIN_ALIAS(SpacecraftThrusterModel,
+                          "ignition::gazebo::systems::SpacecraftThrusterModel")
```

Comment:
you are targeting `gz-sim8`, you should remove this

Reply:
@ahcorde actually our setup is using GZ Garden, so maybe I need to keep this for now. Unless it is backwards compatible without it?